index.js

```diff
@@ -493,35 +540,32 @@ class K8sExecutor extends Executor {
 
                 return this.breaker.runCommand(statusOptions);
             })
-            .then(resp => {
-                if (resp.statusCode !== 200) {
-                    throw new Error(`Failed to get pod status:${JSON.stringify(resp.body, null, 2)}`);
-                }
-
-                const status = resp.body.status.phase.toLowerCase();
-
-                if (status === 'failed' || status === 'unknown') {
-                    throw new Error(`Failed to create pod. Pod status is:${JSON.stringify(resp.body.status, null, 2)}`);
-                }
-
-                const updateConfig = {
-                    apiUri: this.ecosystem.api,
-                    buildId,
-                    token
-                };
-
-                if (resp.body.spec && resp.body.spec.nodeName) {
-                    updateConfig.stats = {
-                        hostname: resp.body.spec.nodeName,
-                        imagePullStartTime: new Date().toISOString()
-                    };
-                } else {
-                    updateConfig.statusMessage = 'Waiting for resources to be available.';
-                }
-
-                return this.updateBuild(updateConfig).then(() => null);
+            .then(res => {
+                return this.checkPodResponse(res)
+                    .then(() => {
+                        const updateConfig = {
+                            apiUri: this.ecosystem.api,
+                            buildId,
+                            token
+                        };
+
+                        if (res.body.spec && res.body.spec.nodeName) {
+                            updateConfig.stats = {
+                                hostname: res.body.spec.nodeName,
+                                imagePullStartTime: new Date().toISOString()
+                            };
+                        } else {
+                            updateConfig.statusMessage = 'Waiting for resources to be available.';
+                        }
+
+                        return this.updateBuild(updateConfig).then(() => null);
+                    })
+                    .catch(err => {
+                        throw new Error(err);
+                    });
             })
             .then(() => {
+                sleep.msleep(this.podStatusQueryDelay);
```

Comment:
https://www.npmjs.com/package/sleep shouldn't be used. It pauses all JS code. we need an aync sleep best created by Promise & setTimeout. https://www.codegrepper.com/code-examples/delphi/es6+await++for+amount+of+seconds